hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java

```diff
@@ -187,6 +189,60 @@ public static boolean isSegmentPresent(String v2Message, String segmentType) {
 		return false;
 	}
 	
+	/**
+	 * @param v2Message
+	 * @param segmentType
+	 * @return
+	 */
+	public static String getZCBSegment(String v2Message, String segmentType) {
+		String[] v2DataLines_Pharmanet = v2Message.split(LINE_BREAK);
+
+		for (String segment : v2DataLines_Pharmanet) {
+
+			if (segment.startsWith(segmentType)) {
+				String[] messageSegments = segment.split(DOUBLE_BACKSLASH + HL7_DELIMITER);
+				if (messageSegments[0].equalsIgnoreCase(segmentType)) {
+					return segment;
+				}
+			}
+
+		}	
+		
+		return null;
+	}
+	
+	
+	/**
+	 * @param zcbSegment
+	 * ZCB|PharmacyId|DateTime|TraceNumber
+	 * @return Pharmacy id
+	 */
+	public static String getPharmacyId(String zcbSegment) {		
+		if(StringUtils.isNotEmpty(zcbSegment)) {
```

Comment:
Again, consider if isNotBlank() is more safe.

Reply:
done